code/modules/mob/living/carbon/xenomorph/evolution.dm

```diff
@@ -196,6 +195,15 @@
 			to_chat(src, "<span class='warning'>Something in this place is interfering with our link to Queen Mother. We are unable to evolve to a psychic caste here!</span>")
 			return
 
+	else if(new_caste_type == /mob/living/carbon/xenomorph/hivemind) //Special case for dealing with hiveminds - this may be subject to heavy change, such as multiple hiveminds potentially being an option
```

Comment:
trying to special case evolution like this with a mob type is really hacky and a hard no.

Reply:
@spookydonut read above, literally the exact same method as checking for multiple queens and shrikes.

Reply:
`	if(new_caste_type == /mob/living/carbon/xenomorph/queen)
		if(hive.living_xeno_queen) //Do another check after the tick.
			to_chat(src, "<span class='warning'>There already is a Queen.</span>")
			return
	else if(new_caste_type == /mob/living/carbon/xenomorph/shrike)
		if(length(hive.xenos_by_typepath[/mob/living/carbon/xenomorph/shrike]))
			to_chat(src, "<span class='warning'>There already is a Shrike.</span>")
			return
	else if(new_caste_type == /mob/living/carbon/xenomorph/hivemind) //Special case for dealing with hiveminds - this may be subject to heavy change, such as multiple hiveminds potentially being an option
		if(length(hive.xenos_by_typepath[/mob/living/carbon/xenomorph/hivemind]))
			to_chat(src, "<span class='warning'>There cannot be two manifestations of the hivemind's will at once.</span>")
			return`